File: myapp/parent/Step20_Nested_OutputAdapter.cs
using sfn_ut.sfn.nested;
using sfn_ut.myapp.child;

namespace sfn_ut.myapp.parent;

public class Step20_Nested_OutputAdapter : SfnMerge<ParentPayload, ChildPayload> {
  public override ParentPayload Execute(SfnMergeInput<ParentPayload, ChildPayload> input) {
    input.Parent.HappyPathData = input.Child.HappyPathData;
    input.Parent.DataFromChild = input.Child.ServiceChargeInDollars;

    return input.Parent;
  }
}

File: java-sfn-unit-test/src/main/java/com/sfnut/myapp/parent/Step20_Nested_OutputAdapter.java
package com.sfnut.myapp.parent;

import com.sfnut.sfn.nested.SfnMerge;
import com.sfnut.sfn.nested.SfnMergeInput;
import com.sfnut.myapp.child.ChildPayload;

public class Step20_Nested_OutputAdapter extends SfnMerge<ParentPayload, ChildPayload> {
    @Override
    public ParentPayload execute(SfnMergeInput<ParentPayload, ChildPayload> input) {
        input.getParent().setHappyPathData(input.getChild().getHappyPathData());
        input.getParent().setDataFromChild(input.getChild().getServiceChargeInDollars());

        return input.getParent();
    }
}
